scripts/provisioning/cleanupcmd.py

```diff
@@ -47,7 +68,55 @@ def process(self):
     """Main processing function."""
     sys.stdout.write(f"Processing {self.name} {self.url}\n")
     try:
+      # Check if reset phase was performed before this
+      self.detect_if_reset_done()
+
+      # cleanup ldap accounts related to S3
       ldap_action_obj = LdapAccountAction(self.ldap_user, self.ldap_passwd)
       ldap_action_obj.delete_account(self.account_cleanup_dict)
+
+      # cleanup ldap config and schemas
+      self.delete_ldap_config()
+
+      # Erase haproxy configurations
+      self.erase_haproxy_config()
+
     except Exception as e:
       raise e
+
+  def detect_if_reset_done(self):
```

Comment:
@sachitanands  Please review this change since the reset phase done check implemented by you.

Reply:
I assume this will be used as skeleton for the functionality and it will be used once I start coding for this logic.

Reply:
Yes